File: backend/src/models/Citizen.js
class Citizen {
    constructor({
        id,
        user_id,
        document_type,
        document_number,
        document_issuer_state,
        birth_date,
        gender,
        marital_status,
        mother_name,
        address_street,
        address_number,
        address_complement,
        address_neighborhood,
        address_city,
        address_state,
        address_zip_code,
        address_country,
        created_at,
        updated_at,
    }) {
        this.id = id;
        this.user_id = user_id;
        this.document_type = document_type;
        this.document_number = document_number;
        this.document_issuer_state = document_issuer_state;
        this.birth_date = birth_date;
        this.gender = gender;
        this.marital_status = marital_status;
        this.mother_name = mother_name;
        this.address_street = address_street;
        this.address_number = address_number;
        this.address_complement = address_complement;
        this.address_neighborhood = address_neighborhood;
        this.address_city = address_city;
        this.address_state = address_state;
        this.address_zip_code = address_zip_code;
        this.address_country = address_country;
        this.created_at = created_at;
        this.updated_at = updated_at;
    }

    /**
     * Get full address
     */
    getFullAddress() {
        const parts = [
            this.address_street,
            this.address_number,
            this.address_complement,
            this.address_neighborhood,
            this.address_city,
            `${this.address_state} ${this.address_zip_code}`,
            this.address_country,
        ].filter(Boolean);
        return parts.join(', ');
    }

    /**
     * Convert to JSON for response
     */
    toJSON() {
        return {
            id: this.id,
            user_id: this.user_id,
            document_type: this.document_type,
            document_number: this.document_number,
            document_issuer_state: this.document_issuer_state,
            birth_date: this.birth_date,
            gender: this.gender,
            marital_status: this.marital_status,
            mother_name: this.mother_name,
            address: {
                street: this.address_street,
                number: this.address_number,
                complement: this.address_complement,
                neighborhood: this.address_neighborhood,
                city: this.address_city,
                state: this.address_state,
                zip_code: this.address_zip_code,
                country: this.address_country,
            },
        };
    }
}

module.exports = Citizen;

File: taskhub/lib/models/citizen.dart
class Address {
  final String? street;
  final String? number;
  final String? complement;
  final String? neighborhood;
  final String? city;
  final String? state;
  final String? zipCode;
  final String? country;

  Address({
    this.street,
    this.number,
    this.complement,
    this.neighborhood,
    this.city,
    this.state,
    this.zipCode,
    this.country,
  });

  String get fullAddress {
    final parts = [
      street,
      number,
      complement,
      neighborhood,
      city,
      if (state != null && zipCode != null) '$state $zipCode',
      country,
    ].where((p) => p != null && p.isNotEmpty).toList();
    return parts.join(', ');
  }

  factory Address.fromJson(Map<String, dynamic> json) {
    return Address(
      street: json['street'] ?? json['address_street'],
      number: json['number'] ?? json['address_number'],
      complement: json['complement'] ?? json['address_complement'],
      neighborhood: json['neighborhood'] ?? json['address_neighborhood'],
      city: json['city'] ?? json['address_city'],
      state: json['state'] ?? json['address_state'],
      zipCode: json['zip_code'] ?? json['address_zip_code'],
      country: json['country'] ?? json['address_country'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'street': street,
      'number': number,
      'complement': complement,
      'neighborhood': neighborhood,
      'city': city,
      'state': state,
      'zip_code': zipCode,
      'country': country,
    };
  }
}

class Citizen {
  final int? id;
  final int userId;
  final String documentType;
  final String documentNumber;
  final String? documentIssuerState;
  final DateTime? birthDate;
  final String? gender;
  final String? maritalStatus;
  final String? motherName;
  final Address? address;

  Citizen({
    this.id,
    required this.userId,
    required this.documentType,
    required this.documentNumber,
    this.documentIssuerState,
    this.birthDate,
    this.gender,
    this.maritalStatus,
    this.motherName,
    this.address,
  });

  factory Citizen.fromJson(Map<String, dynamic> json) {
    return Citizen(
      id: json['id'],
      userId: json['user_id'] ?? 0,
      documentType: json['document_type'] ?? '',
      documentNumber: json['document_number'] ?? '',
      documentIssuerState: json['document_issuer_state'],
      birthDate: json['birth_date'] != null
          ? DateTime.parse(json['birth_date'])
          : null,
      gender: json['gender'],
      maritalStatus: json['marital_status'],
      motherName: json['mother_name'],
      address: json['address'] != null
          ? Address.fromJson(json['address'])
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'user_id': userId,
      'document_type': documentType,
      'document_number': documentNumber,
      'document_issuer_state': documentIssuerState,
      'birth_date': birthDate?.toIso8601String(),
      'gender': gender,
      'marital_status': maritalStatus,
      'mother_name': motherName,
      'address': address?.toJson(),
    };
  }
}
